### ISSUE TITLE
Too many open files on Ubuntu 24.04

### ISSUE THREAD

**Issue by Kr3m** [Garux/netradiant-custom#193]
I'm getting the following error when trying to compile a map:

```
--- LoadMapFile ---
Script file /home/kevin/.q3a/baseq3/maps/testmap.map was not found: Too many open files
************ ERROR ************
```

**Comment by Garux** [comment]
Likely too much .pk3s, they all are kept open. You can increase this limit in linux.

**Comment by ensiform** [comment]
There are ways to increase max file handles on Windows and Linux for sure via code in main program, not sure about bsds or macOS.

**Comment by Garux** [comment]
I do set maximal available limit in windows, in linux it must be set systemwide from what i read.

**Comment by Kr3m** [comment]
> There are ways to increase max file handles on Windows and Linux for sure via code in main program, not sure about bsds or macOS. And some idtech forks have utilized these. I don't remember the exact functions off the top of my head currently.

Yeah, I was able to get it working by setting it from the terminal and running it from the same terminal.  I think I'll just adjust it system wide.  Just not sure what the safest setting would be, but I guess 4096 should be okay...

**Comment by ensiform** [comment]
> > There are ways to increase max file handles on Windows and Linux for sure via code in main program, not sure about bsds or macOS. And some idtech forks have utilized these. I don't remember the exact functions off the top of my head currently.
> 
> Yeah, I was able to get it working by setting it from the terminal and running it from the same terminal.  I think I'll just adjust it system wide.  Just not sure what the safest setting would be, but I guess 4096 should be okay...

Jk2mv has this code 

https://github.com/mvdevs/jk2mv/blob/master/src%2Fsys%2Fsys_unix.cpp#L572-L589
